src/prefect/engine/serializers/base.py

```diff
@@ -0,0 +1,78 @@
+import base64
+import json
+from typing import Any, Union
+
+import cloudpickle
+
+
+class Serializer:
```

Comment:
Should this be named something more informative like `BytesSerializer` or is that too verbose?

Reply:
My intent with naming it like this was to make it clear this is a base class (can handle just about anything, in theory) but I'm happy to make that more explicit change

Reply:
All Serializers serialize to/from bytes, so IMO `BytesSerializer` doesn't really make sense. `PickleSerializer` might be more explicit, but I'm fine with leaving things as is as well.